### PR TITLE
[146] Allow students to leave groups

### LINKED ISSUE #146: Students should be able to leave non-frozen groups

### PR DESCRIPTION
**Pull request by orenyk**
Resolves #146
- Add :leave route and policy for groups in draws

### CODE REVIEW

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -76,6 +76,16 @@ def reject_pending
     handle_action(path: draw_group_path(@draw, @group), **result)
   end
 
+  def leave
+    membership = current_user.membership
+    result = if membership
```
Comment (esoterik) [inline app/controllers/groups_controller.rb]: why is this necessary if we check for this in the policy?

Comment (orenyk) [inline app/controllers/groups_controller.rb]: Hmm, that's a good question. I guess we don't - if the current user is in `group.members` it _must_ have an accepted membership. I'll update this, thanks!

**Comment by orenyk** [comment]
@esoterik ended up tweaking the MembershipDestroyer a bit, want to take a quick look?

**Comment by esoterik** [comment]
ready to rebase and merge!

**Comment by orenyk** [comment]
Rebased, will merge after green build
